File: 조선빈/Lv.1/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    answer = []
    min_num = 0
    add = 0
    max_num = 0
    i=0
    while i<=5:
        if lottos[i] in win_nums:
            min_num += 1
        if lottos[i] == 0:
            add += 1
        i += 1
    max_num = min_num + add
        
    if max_num == 2:
        win_max = 5
    elif max_num == 3:
        win_max = 4
    elif max_num == 4:
        win_max = 3
    elif max_num == 5:
        win_max = 2
    elif max_num == 6:
        win_max = 1
    else:
        win_max = 6
    answer.append(win_max)
    
    if min_num == 2:
        win_min = 5
    elif min_num == 3:
        win_min = 4
    elif min_num == 4:
        win_min = 3
    elif min_num == 5:
        win_min = 2
    elif min_num == 6:
        win_min = 1
    else:
        win_min = 6    
    answer.append(win_min)

    return answer

File: 김용성/Lv1/로또의 최고 순위와 최저 순위.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <utility>

using namespace std;

vector<int> solution(vector<int> lottos, vector<int> win_nums) {
    
    int cnt = 0;
    for(int i = 0; i < lottos.size(); i++)
    {
        if(lottos[i] == 0)
        {
            cnt ++;
        }
    }
    printf("%d\n",cnt);
    
    for(int i = 0; i < win_nums.size(); i++)
    {
        printf("%d ",win_nums[i]);
    }
    printf("\n");
    
    sort(win_nums.begin(),win_nums.end());
    for(int i = 0; i < win_nums.size(); i++)
    {
        printf("%d ",win_nums[i]);
    }
    printf("\n");
    int rank = 0;
    
    for(int i = 0; i < win_nums.size(); i++)
    {
        if(binary_search(win_nums.begin(), win_nums.end(), lottos[i]))
        {
            rank++;
        }
    }
    printf("%d\n",rank);
    vector<int> answer;
    
    
    
    if(rank +cnt == 6) answer.push_back(1);
    else if(rank+cnt == 5) answer.push_back(2);
    else if(rank+cnt == 4) answer.push_back(3);
    else if(rank+cnt == 3) answer.push_back(4);
    else if(rank+cnt == 2) answer.push_back(5);
    else  answer.push_back(6);
    
    if(rank == 6) answer.push_back(1);
    else if(rank == 5) answer.push_back(2);
    else if(rank == 4) answer.push_back(3);
    else if(rank == 3) answer.push_back(4);
    else if(rank == 2) answer.push_back(5);
    else answer.push_back(6);

    return answer;
}
